### PR TITLE
[core] Added support for translating default system channels

### LINKED ISSUE #3623: add (german) translations for system channel types
System channel labels are currently only available in English. There are bindings that expect German texts, so this should be added (further languages upon request I would say).

See https://github.com/openhab/openhab2-addons/issues/1668.

### PR DESCRIPTION
**Pull request by cweitkamp**
* Added german translations

Closes #3623 

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

**Comment by cweitkamp** [comment]
The current implementation works, but I am not really happy with it. Two points:

1. I copied a lot code from the [XmlChannelTypeProvider](https://github.com/eclipse/smarthome/blob/master/bundles/core/org.eclipse.smarthome.core.thing.xml/src/main/java/org/eclipse/smarthome/core/thing/xml/internal/XmlChannelTypeProvider.java) class into the `DefaultSystemChannelTypeProvider` class. Both classes implements the `ChannelTypeProvider` interface, so maybe it makes sense to create a new abstract `LocalizedChannelTypeProvider` subclass containing the business logic for translating channels, states and more.
2. This is more a general architecture design question: Why are the default system channels hard coded into their own class? Isn't it more useful to define a xml-file for them? Like in every binding? If so, the first point is obsolete.

**Comment by sjka** [comment]
wrt 1.: see #3423 (i.e. [AbstractXmlBasedProvider](https://github.com/sjka/smarthome/blob/85699470184f1649a1781c4e3a4f9ca14a3e47e8/bundles/config/org.eclipse.smarthome.config.xml/src/main/java/org/eclipse/smarthome/config/xml/AbstractXmlBasedProvider.java)) 😄

File: bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties [hunk 1]
```diff
@@ -0,0 +1,6 @@
+channel-type.system.signal-strength.label = Signalst�rke
+channel-type.system.low-battery.label = Niedriger Batteriestatus
+channel-type.system.battery-level.label = Batterie Level
```
Comment (sjka) [inline bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties]: Hmmm, that somehow sounds "Denglisch" to me. The internet comes up with "Batteriezustand" or "Batterieladezustand". Not sure if they are better...

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/DefaultSystemChannelTypeProvider.java [hunk 2]
```diff
@@ -89,34 +100,107 @@
                     new EventOption(CommonTriggerEvents.LONG_PRESSED, null))),
             null);
 
+    private class LocalizedChannelTypeKey {
```
Comment (sjka) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/DefaultSystemChannelTypeProvider.java]: Could you make this a static class? Then I think you also don't need to check for the outer type in hashcode and equals.

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/DefaultSystemChannelTypeProvider.java [hunk 3]
```diff
@@ -89,34 +100,107 @@
...
+        public String locale;
```
Comment (sjka) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/DefaultSystemChannelTypeProvider.java]: final

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/DefaultSystemChannelTypeProvider.java [hunk 4]
```diff
@@ -89,34 +100,107 @@
...
+        public UID uid;
```
Comment (sjka) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/DefaultSystemChannelTypeProvider.java]: final

File: bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties (hunk 1 above)
Comment (cweitkamp) [inline bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties]: What about "Batterieladung"?

**Comment by cweitkamp** [comment]
@sjka Thanks for your review. I addressed all your comments. I will hold my changes until #3423 is merged.

File: bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties (hunk 1 above)
Comment (sjka) [inline bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties]: 👍

**Comment by sjka** [comment]
Thanks!

**Pull request merged**